### ISSUE TITLE
How to use pamusb-agent's events to trigger an "xdotool" command

### ISSUE THREAD

**Issue by GameLostException** [mcdope/pam_usb#226]
### Which version of pam_usb are you running?

0.8.3

### Which distribution are you using?

Arch, Kernel: 6.6.12-1-lts

### Which login manager and desktop environment are you using?

Login mgr and desktop: Xfce4, Windows mgr: Awesome

### What happened?

Hi, firstly thanks for this great tool, it makes my daily work-life much easier.

What I can't figure out though is a way to have the `pamusb-agent`'s `lock` event to trigger a key-press in my env. 
In my use-case, I listen to music via a web-based app tooled with a keyboard media key listener so it is responsive to physical media-key presses, regardless of the active window/desktop, as long as it is pressed within my active session.

I am using `xdotool` to trigger a music stop event programatically. The following command works as expected for instance, as in it effectively stop music play when executed in a shell:
`xdotool key XF86AudioStop`

So I added it into my `/etc/security/pam_usb.conf` config as:
```
...
<agent event="lock">
      <cmd>xdotool key XF86AudioStop</cmd>
      <cmd>xfce4-screensaver-command --lock</cmd>
</agent>
...
```

Upon USB device removal, the screensaver kicks in as expected, but the music still plays.

`journalctl` gives:
```
Jan 30 14:44:07 boris-st pamusb-agent[128284]: Device "BKey" has been removed, locking down user "foo"...
Jan 30 14:44:07 boris-st pamusb-agent[128284]: Running "xfce4-screensaver-command --lock" (as 1000:1000)
Jan 30 14:44:07 boris-st pamusb-agent[128284]: Process exit code: 0
Jan 30 14:44:07 boris-st pamusb-agent[128284]: Process stdout:
Jan 30 14:44:07 boris-st pamusb-agent[128284]: Process stderr:
Jan 30 14:44:07 boris-st pamusb-agent[128284]: Locked.
```
The xdotool bit seems to be ignored, not sure why, any hint you could provide?

### Output of "pamusb-check --debug `whoami`"

```shell
* Authentication request for user "foo" (pamusb-check)
[src/local.c:227] Checking whether the caller (pamusb-check) is local or not...
[src/local.c:238] 	Checking pid 993885 (pamusb-check)...
[src/local.c:238] 	Checking pid 807818 (/usr/bin/zsh)...
[src/local.c:238] 	Checking pid 807807 (kitty)...
[src/local.c:238] 	Checking pid      1 (/sbin/init)...
[src/local.c:296] 	Using DISPLAY :0.0 for utmp search
[src/local.c:301] 	DISPLAY contains screen, truncating...
[src/local.c:050] 	No utmp entry found for tty ":0."
[src/local.c:312] 	Trying to get tty from display server
[src/local.c:322] 		Failed, no result while trying to get TTY from display server
[src/local.c:327] 	Trying to get tty by DISPLAY
[src/local.c:335] 		Failed, no result while searching utmp for display :0. owned by user foo
[src/local.c:344] 	Trying to get tty by loginctl
[src/local.c:203] 		Got tty: tty1
[src/local.c:350] 	Retrying with tty tty1, obtained by loginctl, for utmp search
[src/local.c:053] 		utmp entry for tty "tty1" found
[src/local.c:054] 			utmp->ut_pid: 809
[src/local.c:055] 			utmp->ut_user: foo
[src/local.c:071] 		Checking utmp->ut_addr_v6[0]
[src/local.c:071] 		Checking utmp->ut_addr_v6[1]
[src/local.c:071] 		Checking utmp->ut_addr_v6[2]
[src/local.c:071] 		Checking utmp->ut_addr_v6[3]
[src/local.c:075] 	utmp check successful, request originates from a local source!
[src/local.c:374] No remote access detected, seems to be local request - allowing.
[src/device.c:038] Searching for "BKey" in the hardware database...
* Authentication device "BKey" is connected.
* Performing one time pad verification...
[src/volume.c:098] Searching for volume with uuid 47D8-DA1A.
[src/volume.c:123] Found mount points: /run/media/foo/47D8-DA1A
[src/volume.c:138] Found volume 47D8-DA1A.
[src/volume.c:165] Volume 47D8-DA1A is already mounted.
[src/pad.c:326] Loading device pad...
[src/pad.c:336] Loading system pad...
[src/pad.c:352] Pad match.
[src/pad.c:187] Checking whether pads are expired or not...
[src/pad.c:210] Pads expired 4 seconds ago, updating...
* Regenerating new pads...
[src/pad.c:161] Protecting pad file...
[src/pad.c:161] Protecting pad file...
[src/pad.c:254] Generating 1024 bytes unique pad...
[src/pad.c:272] Writing pad to the device...
[src/pad.c:274] Writing pad to the system...
[src/pad.c:276] Synchronizing filesystems...
[src/pad.c:281] One time pads updated.
* Access granted.
```

### Output of "w"

```shell
14:58:41 up 6 days, 21:14,  1 user,  load average: 2.34, 2.71, 3.16
USER     TTY       LOGIN@   IDLE   JCPU   PCPU  WHAT
foo    tty1      23Jan24  6days  7:24m  0.08s xinit /etc/xdg/xfce4/xinitrc -- /etc/X11/xinit/xserverrc vt1
```

### Output of "loginctl"

```shell
SESSION  UID USER  SEAT  TTY  STATE  IDLE SINCE
      1 1000 foo seat0 tty1 active yes  -

1 sessions listed.
```

**Comment by GameLostException** [comment]
Closing as I overlooked 2 things:
1. needed to actually reboot the agent between 2 changes of the config
2. the agent's logs gave me the solution as in providing the event with the DISPLAY environment: `Process stderr: Error: DISPLAY environment variable is empty. Is there an Xorg or other X server running? You can try setting 'export DISPLAY=:0' and trying again.`

Writing this ticket made me dig further into the issue at least :)

**Comment by mcdope** [comment]
Good ol' rubberducking :D :duck: But: i think its not documented that you need to reload/restart the agent for config changes. So it's good you created this issue, I will check and fix that in case.

Sooo, you still need help? If so: there is a bit about ENV usage in the Wiki, else I'm happy to assist. But at this point all that is left is working out which bits of ENV you need to replicate in the agent config.

**Comment by GameLostException** [comment]
Your wiki is actually where I should have started, I can see now it has a very good examples of the `<env>` element. Regardless, my setup is running all smooth now, no need for further help ATM, but I'll take a rain check just in case :)

So you know, I already spent some time figuring out that the agent was even needed to get these events triggered. The reason is that I discovered pamusb through the [Arch wiki pamusb page](https://wiki.archlinux.org/title/Pam_usb) which did not mention anything about the agent, and was still referring to the original repo your fork is from, with a less detailed wiki. So I took the liberty to add [a section giving some details about it](https://wiki.archlinux.org/title/Pam_usb#Enabling_events_management) and update the repo references to yours instead of Aluzzardi's, this should help the next person following the same path.
